src/io/PipeReader.zig

```diff
@@ -0,0 +1,117 @@
+const bun = @import("root").bun;
+const std = @import("std");
+
+/// Read a blocking pipe without blocking the current thread.
+pub fn PipeReader(
+    comptime This: type,
+    // Originally this was the comptime vtable struct like the below
+    // But that caused a Zig compiler segfault as of 0.12.0-dev.1604+caae40c21
+    comptime getFd: fn (*This) bun.FileDescriptor,
+    comptime getBuffer: fn (*This) *std.ArrayList(u8),
+    comptime onReadChunk: ?fn (*This, chunk: []u8) void,
+    comptime onRegisterPoll: ?fn (*This) void,
+    comptime onDone: fn (*This, []u8) void,
```

Comment:
I feel like this should be renamed to `registerPoll` and `done`

The `on` prefix makes me think its just a callback thats called after the poll is registered, or it is done, but it looks like these functions are the ones actually registering the poll and closing/cleaning this up